src/pages/RFOX/hooks/useTimeInPoolQuery.ts

```diff
@@ -77,66 +64,28 @@ export const getTimeInPoolSeconds = async (
   return BigInt(now) - BigInt(earliestNonZeroStakingBalanceTimestamp)
 }
 
-/**
- * Calculates the time the account has most recently had a non-zero staking balance to now, in seconds
- */
-const fetchTimeInPoolSeconds = async (stakingAssetAccountAddress: string): Promise<bigint> => {
-  const stakeEvent = getAbiItem({ abi: foxStakingV1Abi, name: 'Stake' })
-  const unstakeEvent = getAbiItem({ abi: foxStakingV1Abi, name: 'Unstake' })
-
-  const RFOX_CONTRACT_CREATION_BLOCK_NUMBER = getRfoxContractCreationBlockNumber(
-    RFOX_PROXY_CONTRACT_ADDRESS,
-  )
-
-  try {
-    const [stakeLogs, unstakeLogs] = await Promise.all([
-      client.getLogs({
-        address: RFOX_PROXY_CONTRACT_ADDRESS,
-        event: stakeEvent,
-        fromBlock: RFOX_CONTRACT_CREATION_BLOCK_NUMBER,
-        args: {
-          account: getAddress(stakingAssetAccountAddress),
-        },
-      }),
-      client.getLogs({
-        address: RFOX_PROXY_CONTRACT_ADDRESS,
-        event: unstakeEvent,
-        fromBlock: RFOX_CONTRACT_CREATION_BLOCK_NUMBER,
-        args: {
-          account: getAddress(stakingAssetAccountAddress),
-        },
-      }),
-    ])
-
-    // Sort all logs by block number then log index
-    const sortedLogs = [...stakeLogs, ...unstakeLogs].sort((a, b) => {
-      if (a.blockNumber !== b.blockNumber) return Number(a.blockNumber - b.blockNumber)
-      return Number(a.logIndex - b.logIndex)
-    })
-
-    return getTimeInPoolSeconds(sortedLogs)
-  } catch (error) {
-    console.error(error)
-    throw error
-  }
-}
-
 export const useTimeInPoolQuery = <SelectData = bigint>({
   stakingAssetAccountAddress,
   select,
-}: UseTimeInPoolProps<SelectData>) => {
+}: UseTimeInPoolQueryProps<SelectData>) => {
   const queryKey = useMemo(
     () => ['timeInPool', stakingAssetAccountAddress],
     [stakingAssetAccountAddress],
   )
   const queryFn = useMemo(
     () =>
       stakingAssetAccountAddress
-        ? () => fetchTimeInPoolSeconds(stakingAssetAccountAddress)
+        ? async () => {
+            const sortedAccountLogs = await queryClient.fetchQuery({
+              queryFn: getAccountLogsQueryFn(stakingAssetAccountAddress),
+              queryKey: getAccountLogsQueryKey(stakingAssetAccountAddress),
+            })
+
+            return getTimeInPoolSeconds(sortedAccountLogs)
```

Comment:
stylistic: decouple data-fetching bits and derivation by moving  `getTimeInPoolSeconds(sortedAccountLogs)` to a selector

Reply:
cannot do because `getTimeInPoolSeconds` is async and react-query requires synchronous selectors 😔